native/src/utils/renderHtml.ts

```diff
@@ -71,34 +84,143 @@ const renderJS = (cacheDictionary: Record<string, string>) => `
+      const textNode = document.createTextNode(text)
+      if (className) {
+        element.classList.add(className)
+      }
+      element.appendChild(textNode)
+    }
+
+    function showSettingsButton(element) {
+      function onClickHandler () {
+        window.ReactNativeWebView.postMessage(JSON.stringify({ type: '${SETTINGS_MESSAGE_TYPE}', openSettings: true }))
+      }
+
+      const buttonLabel = ${JSON.stringify(t('layout:settings'))}
+      const button = document.createElement('button')
+      button.name = 'opt-in-settings-button'
+      button.innerHTML = buttonLabel
+      button.id = 'opt-in-settings-button'
+      button.onclick = onClickHandler
+      element.appendChild(button)
+    }
+
+    function showMessageWithSettings(text, iframeContainer) {
+      showMessage(text, iframeContainer, 'iframe-info-text')
+      showSettingsButton(iframeContainer)
+    }
+
+    function showOptIn(text, iframeContainer, source) {
+      function onClickHandler () {
+        window.ReactNativeWebView.postMessage(JSON.stringify({ type: '${IFRAME_MESSAGE_TYPE}', allowedSource: { type: source, allowed: true } }))
+      }
+
+      const checkbox = document.createElement('input')
+      checkbox.type = 'checkbox'
+      checkbox.name = 'opt-in-checkbox'
+      checkbox.id = 'opt-in-checkbox'
+      checkbox.onclick = onClickHandler
+      const label = document.createElement('label')
+      label.htmlFor = 'opt-in-checkbox'
+      label.appendChild(document.createTextNode(text))
+      iframeContainer.appendChild(label)
+      iframeContainer.appendChild(checkbox)
+    }
+
+    function showBlockMessageWithSettings(text, iframeContainer) {
+      showMessage(text, iframeContainer)
+      showSettingsButton(iframeContainer)
+    }
+
+    function handleAllowedIframeSources(iframe, allowedExternalSourcePermissions, iframeSource, iframeContainer) {
+      if (allowedExternalSourcePermissions.some(source => source.type === iframeSource && source.allowed)) {
+        // Add do not track parameter (only working for vimeo)
+        if (iframeSource.toLowerCase() === 'vimeo') {
+          const url = new URL(iframe.src)
+          url.searchParams.append('dnt', '1')
+          iframe.setAttribute('src', url.href)
+        }
+        const message = ${JSON.stringify(
+          t('remoteContent:knownResourceContentMessageOne'),
+        )} + iframeSource + '. ' + ${JSON.stringify(t('remoteContent:knownResourceContentMessageTwo'))}
+          showMessageWithSettings(message, iframeContainer)
+      } else if (allowedExternalSourcePermissions.some(source => source.type === iframeSource && !source.allowed)) {
+          const translation = ${JSON.stringify(t('remoteContent:knownResourceBlocked'))}
+          const message = iframeSource + ' ' + translation
+          showBlockMessageWithSettings(message, iframeContainer)
+          iframe.remove()
+      }
+      // No permissions set for listed sources
```

Comment:
Should be clear enough by the code, e.g. by changing the order and using the suggestion above
```suggestion
```
E.g.
``` ts
      const supportedExternalSourcePermission = allowedExternalSourcePermissions.find(source => source.type === iframeSource)
      if (!supportedExternalSourcePermission) {
        // show opt in
      } else if (supportedExternalSourcePermission.allowed) {
        // show iframe
      } else {
        // show blocked message
      }
```